tests/e2e/test_e2e.py

```diff
@@ -57,6 +57,16 @@ def test_run():
     os.remove("client.py")
 
 
+@e2e_from_file("tests/simple_server_diff_port.py")
+def test_run():
+    assert os.path.exists("client.py"), f"Expected client file to be created at {os.getcwd()} after starting the server"
+    output = subprocess.run("python client.py", shell=True, capture_output=True, text=True).stdout
+    assert (
```

Comment:
can you please also read the generated `client.py` and assert that the port is `8080` there?